build fails on Mac OS X

After `make cleanall`.  My last successful build was at 1fd440e.
```
libgit2/libgit2.jl
LoadError("sysimg.jl",324,LoadError("libgit2/libgit2.jl",13,LoadError("libgit2/consts.jl",282,ErrorException("error compiling version: could not load library \"libgit2\"\ndlopen(libgit2.dylib, 1): Library not loaded: libssh2.1.dylib\n  Referenced from: /Users/sachs/src/julia-master/usr/lib//libgit2.dylib\n  Reason: image not found"))))
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/sachs/src/julia-master/usr/lib/julia/sys.o] Error 1
make: *** [julia-sysimg-release] Error 2
```

--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -19,8 +19,6 @@ else
 LIBGIT2_OPTS += -DBUILD_CLAR=OFF -DDLLTOOL=`which $(CROSS_COMPILE)dlltool`
 LIBGIT2_OPTS += -DCMAKE_FIND_ROOT_PATH=/usr/$(XC_HOST) -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY
 endif
-else
-LIBGIT2_OPTS += -DCMAKE_INSTALL_RPATH=$(build_prefix) -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
 endif
 
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/CMakeLists.txt $(LIBSSH2_OBJ_TARGET)
@@ -49,6 +47,7 @@ ifeq ($(BUILD_OS),WINNT)
 else
 	$(call make-install,$(LIBGIT2_SRC_DIR),)
 endif
+	$(INSTALL_NAME_CMD)libgit2.$(SHLIB_EXT) $@
 ifeq ($(OS),Linux)
 	@# If we're on linux, copy over libssl and libcrypto for libgit2
 	-LIBGIT_LIBS=$$(ldd "$@" | tail -n +2 | awk '{print $$(NF-1)}'); \

--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -21,6 +21,10 @@ LIBGIT2_OPTS += -DCMAKE_FIND_ROOT_PATH=/usr/$(XC_HOST) -DCMAKE_FIND_ROOT_PATH_MO
 endif
 endif
 
+ifeq ($(OS),Linux)
+LIBGIT2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
+endif
+
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/CMakeLists.txt $(LIBSSH2_OBJ_TARGET)
 ifeq ($(OS),Linux)
 	-cd $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/libgit2-require-openssl.patch

--- a/deps/libssh2.mk
+++ b/deps/libssh2.mk
@@ -19,6 +19,10 @@ else
 LIBSSH2_OPTS += -DCRYPTO_BACKEND=mbedTLS -DENABLE_ZLIB_COMPRESSION=ON
 endif
 
+ifeq ($(OS),Linux)
+LIBSSH2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
+endif
+
 $(BUILDDIR)/$(LIBSSH2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR)/CMakeLists.txt $(MBEDTLS_OBJ_TARGET)
 	-cd $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/libssh2-mbedtls.patch
 	mkdir -p $(dir $@)

--- a/deps/libssh2.mk
+++ b/deps/libssh2.mk
@@ -8,7 +8,6 @@ LIBSSH2_OBJ_SOURCE := $(BUILDDIR)/$(LIBSSH2_SRC_DIR)/src/libssh2.$(SHLIB_EXT)
 LIBSSH2_OBJ_TARGET := $(build_shlibdir)/libssh2.$(SHLIB_EXT)
 
 LIBSSH2_OPTS := $(CMAKE_COMMON) -DBUILD_SHARED_LIBS=ON -DBUILD_EXAMPLES=OFF \
-				-DCMAKE_INSTALL_RPATH=$(build_prefix) -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE \
 				-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LIBDIR=lib
 
 ifeq ($(OS),WINNT)
@@ -43,6 +42,7 @@ ifeq ($(BUILD_OS),WINNT)
 else
 	$(call make-install,$(LIBSSH2_SRC_DIR),)
 endif
+	$(INSTALL_NAME_CMD)libssh2.$(SHLIB_EXT) $@
 	touch -c $(LIBSSH2_OBJ_TARGET)
 
 clean-libssh2:

--- a/deps/mbedtls.mk
+++ b/deps/mbedtls.mk
@@ -22,6 +22,10 @@ else
 MBEDTLS_OPTS += -DENABLE_ZLIB_SUPPORT=ON
 endif
 
+ifeq ($(OS),Linux)
+MBEDTLS_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
+endif
+
 $(SRCDIR)/srccache/$(MBEDTLS_SRC).tgz: | $(SRCDIR)/srccache
 	$(JLDOWNLOAD) $@ $(MBEDTLS_URL)
 

--- a/deps/mbedtls.mk
+++ b/deps/mbedtls.mk
@@ -11,8 +11,7 @@ MBEDTLS_OBJ_SOURCE := $(BUILDDIR)/mbedtls-$(MBEDTLS_VER)/library/libmbedcrypto.$
 MBEDTLS_OBJ_TARGET := $(build_shlibdir)/libmbedcrypto.$(SHLIB_EXT)
 
 MBEDTLS_OPTS := $(CMAKE_COMMON) -DUSE_SHARED_MBEDTLS_LIBRARY=ON \
-		-DENABLE_PROGRAMS=OFF -DCMAKE_BUILD_TYPE=Release \
-		-DCMAKE_INSTALL_RPATH=$(build_prefix) -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+		-DENABLE_PROGRAMS=OFF -DCMAKE_BUILD_TYPE=Release
 
 ifeq ($(OS),WINNT)
 MBEDTLS_OPTS += -DENABLE_ZLIB_SUPPORT=OFF
@@ -55,6 +54,12 @@ ifeq ($(OS), WINNT)
 else
 	$(call make-install,mbedtls-$(MBEDTLS_VER),)
 endif
+	$(INSTALL_NAME_CMD)libmbedx509.$(SHLIB_EXT) $(build_shlibdir)/libmbedx509.$(SHLIB_EXT)
+	$(INSTALL_NAME_CMD)libmbedtls.$(SHLIB_EXT) $(build_shlibdir)/libmbedtls.$(SHLIB_EXT)
+	$(INSTALL_NAME_CHANGE_CMD) libmbedx509.0.dylib @rpath/libmbedx509.$(SHLIB_EXT) $(build_shlibdir)/libmbedtls.$(SHLIB_EXT)
+	$(INSTALL_NAME_CHANGE_CMD) libmbedcrypto.0.dylib @rpath/libmbedcrypto.$(SHLIB_EXT) $(build_shlibdir)/libmbedtls.$(SHLIB_EXT)
+	$(INSTALL_NAME_CHANGE_CMD) libmbedcrypto.0.dylib @rpath/libmbedcrypto.$(SHLIB_EXT) $(build_shlibdir)/libmbedx509.$(SHLIB_EXT)
+	$(INSTALL_NAME_CMD)libmbedcrypto.$(SHLIB_EXT) $@
 	touch -c $(MBEDTLS_OBJ_TARGET)
 
 clean-mbedtls:

